be/src/exec/pipeline/sort/partition_sort_sink_operator.cpp

```diff
@@ -0,0 +1,119 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/pipeline/sort/partition_sort_sink_operator.h"
+
+#include <execinfo.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "column/chunk.h"
+#include "column/column_helper.h"
+#include "exec/pipeline/sort/sort_context.h"
+#include "exec/vectorized/chunks_sorter.h"
+#include "exec/vectorized/chunks_sorter_full_sort.h"
+#include "exec/vectorized/chunks_sorter_topn.h"
+#include "exprs/expr.h"
+#include "runtime/exec_env.h"
+#include "runtime/runtime_state.h"
+
+using namespace starrocks::vectorized;
+
+namespace starrocks::pipeline {
+Status PartitionSortSinkOperator::prepare(RuntimeState* state) {
+    RETURN_IF_ERROR(Operator::prepare(state));
+    return Status::OK();
+}
+
+Status PartitionSortSinkOperator::close(RuntimeState* state) {
+    return Operator::close(state);
+}
+
+StatusOr<vectorized::ChunkPtr> PartitionSortSinkOperator::pull_chunk(RuntimeState* state) {
+    CHECK(false) << "Shouldn't pull chunk from result sink operator";
+}
+
+bool PartitionSortSinkOperator::need_input() const {
+    return true;
+}
+
+Status PartitionSortSinkOperator::push_chunk(RuntimeState* state, const vectorized::ChunkPtr& chunk) {
+    vectorized::ChunkPtr materialize_chunk = _materialize_chunk_before_sort(chunk.get());
+    RETURN_IF_ERROR(_chunks_sorter->update(state, materialize_chunk));
+    return Status::OK();
+}
+
+vectorized::ChunkPtr PartitionSortSinkOperator::_materialize_chunk_before_sort(vectorized::Chunk* chunk) {
+    vectorized::ChunkPtr materialize_chunk = std::make_shared<vectorized::Chunk>();
+
+    // materialize all sorting columns: replace old columns with evaluated columns
+    const size_t row_num = chunk->num_rows();
+    const auto& slots_in_row_descriptor = _materialized_tuple_desc->slots();
+    const auto& slots_in_sort_exprs = _sort_exec_exprs.sort_tuple_slot_expr_ctxs();
+
+    DCHECK_EQ(slots_in_row_descriptor.size(), slots_in_sort_exprs.size());
+
+    for (size_t i = 0; i < slots_in_sort_exprs.size(); ++i) {
+        ExprContext* expr_ctx = slots_in_sort_exprs[i];
+        ColumnPtr col = expr_ctx->evaluate(chunk);
```

Comment:
These codes are repeated with `SortSinkOperator::_materialize_chunk_before_sort` and `TopNNode::_materialize_chunk_before_sort`

Reply:
Will use a common materialize_chunk_before_sort in ChunksSorter